wannierberri/calculators/static.py

```diff
@@ -188,74 +185,54 @@ class Morb(StaticCalculator):
         | Eq(1) in `Ref <https://journals.aps.org/prb/abstract/10.1103/PhysRevB.85.014435>`_
         | Output: :math:`M = -\int [dk] (H + G - 2E_f \cdot \Omega) f`"""
 
-    def __init__(self, **kwargs):
+    def __init__(self, constant_factor=-factors.eV_au / factors.bohr**2, **kwargs):
         self.Formula = frml.Morb_Hpm
-        self.factor = 1
         self.fder = 0
-        super().__init__(**kwargs)
+        super().__init__(constant_factor=constant_factor, **kwargs)
+        self.AHC = AHC(constant_factor=constant_factor, print_comment=False, **kwargs)
 
     def __call__(self, data_K):
         Hplus_res = super().__call__(data_K)
-        Omega_res = AHC(Efermi=self.Efermi, tetra=self.tetra, smoother=self.smoother,
-                use_factor=False, print_comment=False,
-                kwargs_formula=self.kwargs_formula)(data_K).mul_array(self.Efermi)
-        final_factor = -factors.eV_au / factors.bohr**2
-        if not self.use_factor:
-            final_factor = np.sign(final_factor)
-        #with use_factor=False, the factor of AHC is -1, so it is '+' below.
-        return final_factor * data_K.cell_volume * (Hplus_res + 2 * Omega_res)
+        Omega_res = self.AHC(data_K).mul_array(self.Efermi)
+        return (Hplus_res - 2 * Omega_res) * data_K.cell_volume
```

Comment:
I simplified this pattern (additional calculator in `__call__`) by initiating the calculator in `__init__` and storing it as an attribute.